packages/odyssey-storybook/src/components/odyssey-mui/MenuButton/MenuButton.stories.tsx

```diff
@@ -176,6 +176,34 @@ export const Simple: StoryObj<MenuButtonProps> = {
   },
 };
 
+/**
+ * Note that the falsy children in the following MenuButton won't appear in the source
+ * code in the actual story in your browser, but this illustrates that there are no
+ * compile-time issues with the falsy items passed as children.
+ */
+export const FalsyChildren: StoryObj<MenuButtonProps> = {
+  args: {
+    buttonLabel: "More actions",
+    children: [
+      <MenuItem key="1">Truthy Menu Item</MenuItem>,
+      false,
+      null,
+      undefined,
+    ],
+  },
+  play: async ({
+    args,
+    canvasElement,
+    step,
+  }: {
+    args: MenuButtonProps;
+    canvasElement: HTMLElement;
+    step: PlaywrightProps<MenuButtonProps>["step"];
+  }) => {
+    clickMenuButton({ canvasElement, step })(args, "Menu Button Simple");
+  },
+};
+
```

Comment:
I don't think we want a story for this. Seems like a unit test is enough, but we're unit testing our types right? It always accepted falsy children just that TS complained.

We shouldn't be adding unit tests for type checks. We should expect TS works as-expected right?

Unless we want to add TS-specific unit tests, then those should happen outside of Storybook because we'd be verifying that the types we defined work as we expect.

We don't want people to pass no items, it's just that sometimes they do right?

I can't remember the previous reason we added that `NullElement`.

Why would someone pass `null` to this component? There was a reason for it though. Shouldn't they just wrap the whole `MenuButton` in a condition instead of passing it nothing? `items.length > 0`?

Reply:
> It always accepted falsy children just that TS complained.

It worked in JS with falsy children because the type-checking no longer applies. I don't believe it's a good dev-experience though for consumers to get TS errors and need to squash them with a `ts-ignore` because our typing is wrong and possibly leading to build failures on their end.

> We shouldn't be adding unit tests for type checks. We should expect TS works as-expected right?

Just because TS is happy with the type, doesn't mean its typed to our expectations, hence the bug that I am addressing here.

I'm OK with removing this and I agree that a story may not be the right place for a compile-time test to prove that the type works as expected. I wanted a way to show though, that a) this works and b) that it doesn't break again.

Is there a better way going forward to include a regression test for something like this?